src/operators/Aggregate.cpp

```diff
@@ -0,0 +1,515 @@
+#include "Aggregate.h"
+
+#include <utility>
+#include <arrow/api.h>
+#include <arrow/compute/api.h>
+
+#include <table/util.h>
+#include <iostream>
+
+
+namespace hustle::operators {
+
+    Aggregate::Aggregate(
+            std::shared_ptr<OperatorResult> prev_result,
+            std::vector<AggregateReference> aggregate_refs,
+            std::vector<ColumnReference> group_refs,
+            std::vector<ColumnReference> order_by_refs) {
+
+        prev_result_ = prev_result;
+        aggregate_refs_ = aggregate_refs;
+
+        group_by_refs_ = std::move(group_refs);
+        order_by_refs_ = std::move(order_by_refs);
+
+        aggregate_builder_ = get_aggregate_builder(aggregate_refs_[0].kernel);
+
+        // Fetch the fields associated with each groupby column.
+        std::vector<std::shared_ptr<arrow::Field>> group_by_fields;
+        for(auto &group_by : group_by_refs_) {
+            group_by_fields.push_back(
+                    group_by.table->get_schema()
+                    ->GetFieldByName(group_by.col_name));
+        }
+
+        // Initialize a StructBuilder containing one builder for each group
+        // by column.
+        group_type = arrow::struct_(group_by_fields);
+        group_builder = std::make_shared<arrow::StructBuilder>(
+                group_type, arrow::default_memory_pool(), get_group_builders());
+
+        // Fetch unique values for all Group By columns. You need to do fetch
+        // all of them at once, since you need to know how many unique values
+        // are in each column to initialize the loop variables.
+        for (auto &col_ref : group_by_refs_) {
+            unique_values_.push_back(get_unique_values(col_ref));
+        }
+
+    }
+
+
+    std::vector<std::shared_ptr<arrow::ArrayBuilder>>
+    Aggregate::get_group_builders() {
```

Comment:
Could we add comments to the functions so that it is easier to read? I think it will also help you @NicholasCorrado when you go back to this code in few months :-) I'm looking to you, @johnchronis and @gaffneyk to lead on the documentation-by-example so we can get a good coding culture in place.